src/main/cpp/src/get_json_object.cu

```diff
@@ -70,14 +73,49 @@ struct path_instruction {
   cudf::string_view name;
 
   // used when type is index
-  int index{-1};
+  int64_t index{-1};
 };
 
-// TODO parse JSON path
-thrust::optional<rmm::device_uvector<path_instruction>> parse_path(
-  cudf::string_scalar const& json_path)
+rmm::device_uvector<path_instruction> construct_path_commands(cudf::table_view const& instructions,
```

Comment:
Why is the input to this a table? We only support a scalar value as input for the path. The data is small, so we are going to launch a kernel to parse a list of strings on the GPU, when the CPU could do it and send the results to the GPU faster than this. Please change it so that we have a list/vector of C++ objects that represent the parsed data. This cannot be a string because I may want to use the word `star`, `key` or really any other of the special case words you have here in my path, and I don't want it to fail. The magic numbers should not pass through into the cuda kernel, they should only be between the java and the JNI implementation. Java should have an enum for the type, and there should be a corresponding enum in C++. The java code should translate the enum to an int, or byte in this case because there are not many values.

Reply:
If the input is a small number of instructions (?), let just do all processing on the CPU using host code, or even call the Spark's code to do such preprocessing. Then, we can create a `device_uvector` from the result using `make_device_uvector_sync` for further GPU processing.

Reply:
Thanks all! updated to matched enum and struct in cpu.